ambari-logsearch-server/src/main/java/org/apache/ambari/logsearch/manager/MetadataManager.java

```diff
@@ -119,36 +119,36 @@ public void deleteMetadata(String id) {
   }
 
   @SuppressWarnings("unchecked")
-  public LogSearchMetaDataListResponse getMetadata(EventHistoryRequest request) {
+  public LogSearchMetaDataListResponse getMetadata(MetadataRequest request) {
     LogSearchMetaDataListResponse response = new LogSearchMetaDataListResponse();
     String rowType = request.getRowType();
     if (StringUtils.isBlank(rowType)) {
       throw new MalformedInputException("Row type was not specified");
     }
 
-    SolrQuery evemtHistoryQuery = conversionService.convert(request, SolrQuery.class);
-    evemtHistoryQuery.addFilterQuery(String.format("%s:%s OR %s:%s", USER_NAME, LogSearchContext.getCurrentUsername(),
+    SolrQuery metadataQueryQuery = conversionService.convert(request, SolrQuery.class);
+    metadataQueryQuery.addFilterQuery(String.format("%s:%s OR %s:%s", USER_NAME, LogSearchContext.getCurrentUsername(),
         SHARE_NAME_LIST, LogSearchContext.getCurrentUsername()));
-    SolrDocumentList solrList = metadataSolrDao.process(evemtHistoryQuery).getResults();
+    SolrDocumentList solrList = metadataSolrDao.process(metadataQueryQuery).getResults();
 
     Collection<LogsearchMetaData> configList = new ArrayList<>();
 
     for (SolrDocument solrDoc : solrList) {
-      LogsearchMetaData eventHistoryData = new LogsearchMetaData();
-      eventHistoryData.setFiltername("" + solrDoc.get(FILTER_NAME));
-      eventHistoryData.setId("" + solrDoc.get(ID));
-      eventHistoryData.setValues("" + solrDoc.get(VALUES));
-      eventHistoryData.setRowType("" + solrDoc.get(ROW_TYPE));
+      LogsearchMetaData metadata = new LogsearchMetaData();
+      metadata.setFiltername("" + solrDoc.get(FILTER_NAME));
```

Comment:
1. `String.valueOf(solrDoc.get(FILTER_NAME))` can be used here.
2. If `solrDoc.get(FILTER_NAME)` returns null the result string will be `"null"`. Is it intended?

Reply:
same as above. also note that part will be the next to rewrite as Tobi will start to use metadata collections for storing user settings.

Reply:
also filterName is required field in Solr so we won't get null here